### PR TITLE
Change reported metrics for if-eval

### PR DESCRIPTION
**Pull request by Kipok**
Changed the reported to standard prompt/instruction strict/loose accuracies. Tested that our numbers roughly match what's in open llm leaderboard

### CODE REVIEW

File: pipeline/compute_metrics.py [hunk 1]
```diff
@@ -140,9 +141,37 @@ def is_incomplete(self, elem):
         if incomplete:
             return False
```
Comment (shtoshni) [inline pipeline/compute_metrics.py:142]: Should it be True or False?

Comment (Kipok) [reply pipeline/compute_metrics.py:142]: Good catch, should be True

File: pipeline/compute_metrics.py [hunk 2]
```diff
@@ -138,11 +139,39 @@ def fill_up_missing(self):
...
         incomplete = 'loose_eval' not in elem or 'strict_eval' not in elem
         if incomplete:
-            return False
+            return True
         return (
-            'follow_all_instructions' not in elem['loose_eval'] or 'follow_all_instructions' not in elem['strict_eval']
+            'follow_instruction_list' not in elem['loose_eval']
+            or 'follow_instruction_list' not in elem['strict_eval']
+            and 'instruction_id_list' not in elem['loose_eval']
+            or 'instruction_id_list' not in elem['strict_eval']
```
Comment (shtoshni) [inline pipeline/compute_metrics.py:147]: This is a complex clause imo. There is "or" and "and" mixed in without parenthesis which can make it hard to parse the true condition. I think you're trying to cover for loose and strict eval conditions.  

This might be a cleaner way to get to it:

if 'follow_instruction_list' in elem['loose_eval'] and 'instruction_id_list' in elem['loose_eval']:
     return False
if 'follow_instruction_list' in elem['strict_eval'] and 'instruction_id_list' in elem['strict_eval']:
     return False

return True

Comment (Kipok) [reply pipeline/compute_metrics.py]: thanks, in fact there were even more mistakes related to that logic, hopefully it's correct now

**Review by shtoshni — APPROVED** [review]
